drivers/interrupt_controller/intc_plic.c

```diff
@@ -337,6 +356,117 @@ static int plic_init(const struct device *dev)
 	return 0;
 }
 
+#ifdef CONFIG_PLIC_SHELL
+static inline int parse_device(const struct shell *sh, size_t argc, char *argv[],
+			       const struct device **plic)
+{
+	ARG_UNUSED(argc);
+
+	*plic = device_get_binding(argv[1]);
+	if (!*plic) {
+		shell_error(sh, "PLIC device (%s) not found!\n", argv[1]);
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int cmd_stats(const struct shell *sh, size_t argc, char *argv[])
+{
+	const struct device *dev;
+	int ret = parse_device(sh, argc, argv, &dev);
```

Comment:
Maybe as a follow-up PR (or can be done here too), would be nice to be able to specify some additional arguments.

Positional arguments could just be irqs (so print only the named irqs, otherwise all).

`plic stats get <device> [irq..]`

Additional non-positional arguments could be e.g `--min count` to only print irqs greater than a `count`.

A bit of a pipe dream, but it would be nice to print the human readable device ids associated with each irq (e.g. `--name` or similar option). I know labels went the way of the dinosaur, but maybe there is some other way.

`plic stats get <device> --name`

Irq, name, count

Reply:
I've added an optional argument to only print irqs greater than a `count`
```
plic stats get <device> [minimum hits]
```